CHANGELOG.md

```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Multi-style plotting capabilities for generated example plots
 - JSON file for plotting themes
 - Smoke testing in relevant tox environments
+- Basic Transfer Learning example
```

Comment:
[Putting this here such that we can discuss things]

@AdrianSosic @Scienfitz after the rebase, there are in fact only two things still open to discuss:
1. Are we happy with the current content, state and text of the example?
2. Do we want to wait for the backend-fixing PR #130 before merging this one?
My opinion: I am happy with the current content, and I would prefer to first merge this and change the example later during a general refactoring of the examples, or even during #130 as I think we should have this example here available sooner rather than later.

Reply:
Also, just to make sure: Learning curves are in general not aligned for different numbers of data available in the beginning, will this be fixed once the other PR is implemented or is this expected?
![image](https://github.com/emdgroup/baybe/assets/84016733/ce042639-8cb4-412d-af30-f359be62d616)

Reply:
@Scienfitz and I agreed that we will not wait for #130 . Hence, the only question remains: Are we happy with the user guide in its current version?

Reply:
ill resolve this as all has been agreed and im indifferent whether you want to remove or keep the 0% baseline as well